tests/virt/data/disk_devices.xml

```diff
@@ -0,0 +1,161 @@
+  </source>
+  <target bus="virtio" dev="vda"/>
+  <serial>54-a672-23e5b495a9ed</serial>
+  <driver cache="none" error_policy="stop"
+          io="threads" name="qemu" type="raw"/>
+</disk>
+<disk device="lun" sgio="unfiltered" snapshot="no" type="block">
+  <source dev="/dev/mapper/lun1">
+    <seclabel model="dac" relabel="no" type="none" />
+  </source>
+  <target bus="scsi" dev="sda"/>
+  <driver cache="none" error_policy="stop"
+          io="native" name="qemu" type="raw"/>
+</disk>
+<disk device="disk" snapshot="no" type="network">
+  <source name="poolname/volumename" protocol="rbd">
+    <host name="1.2.3.41" port="6789" transport="tcp"/>
+    <host name="1.2.3.42" port="6789" transport="tcp"/>
+  </source>
+  <target bus="virtio" dev="vda"/>
+  <driver cache="none" error_policy="stop"
+          io="threads" name="qemu" type="raw"/>
+</disk>
+<disk device="disk" snapshot="no" type="network">
+  <source name="poolname/volumename" protocol="rbd">
+    <host name="1.2.3.41" port="6789" transport="tcp"/>
+    <host name="1.2.3.42" port="6789" transport="tcp"/>
+  </source>
+  <auth username="cinder">
+    <secret type="ceph" uuid="abcdef"/>
+  </auth>
+  <target bus="virtio" dev="vda"/>
+  <serial>54-a672-23e5b495a9ea</serial>
+  <driver cache="none" error_policy="stop"
+          io="threads" name="qemu" type="raw"/>
+</disk>
+<disk device="lun" sgio="unfiltered" snapshot="no" type="block">
+  <address bus="0" controller="0" target="0" type="drive" unit="0" />
+  <source dev="/dev/mapper/36001405b3b7829f14c1400d925eefebb">
+    <seclabel model="dac" relabel="no" type="none" />
+  </source>
+  <target bus="scsi" dev="sda" />
+  <driver cache="none" error_policy="stop" io="native"
+          name="qemu" type="raw" />
+</disk>
+<disk type='block' device='disk' snapshot='no'>
+  <driver name='qemu' type='raw' cache='none' error_policy='stop' io='native'/>
+  <source dev='/dev/mapper/360014055bd72b7a8c2c4f338cdbdc258' index='2'>
+    <seclabel model='dac' relabel='no'/>
+  </source>
+  <backingStore/>
+  <target dev='sdb' bus='scsi'/>
+  <serial>cca4f09d-fdf7-46b9-9e77-a96e293eb33f</serial>
+  <alias name='ua-cca4f09d-fdf7-46b9-9e77-a96e293eb33f'/>
+  <address type='drive' controller='0' bus='0' target='0' unit='2'/>
+</disk>
+<disk type='block' device='disk' snapshot='no'>
+  <driver name='qemu' type='raw' cache='none' error_policy='stop' io='native'/>
+  <source dev='/dev/mapper/36001405f4f3d2dee8494a21a89d43179' index='1'>
+    <seclabel model='dac' relabel='no'/>
+  </source>
+  <backingStore/>
+  <target dev='sdd' bus='scsi'/>
+  <serial>ed0fb068-840e-4c9f-a1ec-1767de89c0df</serial>
+  <alias name='ua-ed0fb068-840e-4c9f-a1ec-1767de89c0df'/>
+  <address type='drive' controller='0' bus='0' target='0' unit='1'/>
+</disk>
+<disk type='block' device='disk' snapshot='no'>
+  <driver name='qemu' type='raw' cache='none' error_policy='stop' io='native'/>
+  <source dev='/dev/mapper/36001405d6f6b81aa3d742de9af08f906' index='1'>
```

Comment:
Since direct luns must use "<source dev=/dev/mapper/..." - can we replace the lookup
code with xpath finding disks that have such source?

Reply:
Good idea, thank you for the tip!  Done.